pkg/vault/contracts/RouterExtension.sol

```diff
@@ -0,0 +1,414 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IPermit2 } from "permit2/src/interfaces/IPermit2.sol";
+
+import { IWETH } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/misc/IWETH.sol";
+import { IRouterCommon } from "@balancer-labs/v3-interfaces/contracts/vault/IRouterCommon.sol";
+import { IRouterExtension } from "@balancer-labs/v3-interfaces/contracts/vault/IRouterExtension.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import {
+    ReentrancyGuardTransient
+} from "@balancer-labs/v3-solidity-utils/contracts/openzeppelin/ReentrancyGuardTransient.sol";
+
+import { Router } from "./Router.sol";
+import { RouterCommon } from "./RouterCommon.sol";
+
+contract RouterExtension is RouterCommon, ReentrancyGuardTransient, IRouterExtension {
```

Comment:
This one should have a fallback that reverts if it receives ETH.

Reply:
Should be the same situation as the VaultAdmin (end of the Proxy line) - receive that reverts unconditionally (no payables means any calls directly to query methods with ETH would fail too), and the same fallback as VaultAdmin